File: Java/Agenda/app/src/main/java/io/github/henriquemcc/agenda/java/dao/AlunoDAO.java
package io.github.henriquemcc.agenda.java.dao;

import java.util.ArrayList;
import java.util.List;

import io.github.henriquemcc.agenda.java.model.Aluno;

public class AlunoDAO
{
	private final static List<Aluno> alunos = new ArrayList<>();
	private static int contadorDeIds = 1;

	public void salva(Aluno aluno)
	{
		aluno.setId(contadorDeIds);
		alunos.add(aluno);
		atualizaIds();
	}

	private void atualizaIds()
	{
		contadorDeIds++;
	}

	public Aluno buscaAlunoPeloId(Aluno aluno)
	{
		if (aluno != null)
		{
			for (int indice = 0; indice < alunos.size(); indice++)
			{
				if (alunos.get(indice) != null)
				{
					if (alunos.get(indice).getId() == aluno.getId())
					{
						return alunos.get(indice);
					}
				}
			}
		}
		return null;
	}

	public void edita(Aluno aluno) {
		Aluno alunoEncontrado = buscaAlunoPeloId(aluno);
		if (alunoEncontrado != null)
		{
			int posicaoDoAluno = alunos.indexOf(alunoEncontrado);
			alunos.set(posicaoDoAluno, aluno);
		}
	}

	public void remove(Aluno aluno)
	{
		Aluno alunoDevolvido = buscaAlunoPeloId(aluno);
		if (alunoDevolvido != null)
		{
			alunos.remove(alunoDevolvido);
		}
	}

	public List<Aluno> todos()
	{
		return new ArrayList<>(alunos);
	}
}

File: Kotlin/Agenda/app/src/main/java/io/github/henriquemcc/agenda/kotlin/dao/AlunoDAO.kt
package io.github.henriquemcc.agenda.kotlin.dao

import io.github.henriquemcc.agenda.kotlin.model.Aluno

class AlunoDAO
{
	companion object
	{
		private val alunos = ArrayList<Aluno>()
		private var contadorDeIds = 1
	}

	fun salva(aluno: Aluno)
	{
		aluno.id = contadorDeIds
		alunos.add(aluno)
		atualizaIds()
	}

	private fun atualizaIds()
	{
		contadorDeIds++
	}

	private fun buscaAlunoPeloId(aluno: Aluno): Aluno?
	{
		for(index in 0..alunos.size)
		{
			if (aluno.id == alunos[index].id)
			{
				return alunos[index]
			}
		}
		return null
	}

	fun edita(aluno: Aluno)
	{
		val alunoEncontrado = buscaAlunoPeloId(aluno)

		if (alunoEncontrado != null)
		{
			val posicaoDoAluno = alunos.indexOf(alunoEncontrado)
			alunos.set(posicaoDoAluno, aluno)
		}
	}

	fun remove(aluno: Aluno)
	{
		val alunoDevolvido = buscaAlunoPeloId(aluno)
		if (alunoDevolvido != null)
		{
			alunos.remove(alunoDevolvido)
		}
	}

	fun todos(): List<Aluno>
	{
		return ArrayList(alunos)
	}
}